src/RamOperation.h

```diff
@@ -606,7 +696,10 @@ class RamIndexChoice : public RamIndexOperation, public RamAbstractChoice {
 
     bool equal(const RamNode& node) const override {
         const auto& other = static_cast<const RamIndexChoice&>(node);
-        return RamIndexOperation::equal(other) && RamAbstractChoice::equal(other);
+        return RamRelationOperation::equal(other) &&
+               equal_targets(queryPattern.first, other.queryPattern.first) &&
+               equal_targets(queryPattern.second, other.queryPattern.second) &&
+               getCondition() == other.getCondition();
```

Comment:
The version from master looks better
```suggestion
        return RamIndexOperation::equal(other) && RamAbstractChoice::equal(other);
```

Reply:
Actually, the static cast isn't needed either.